### ISSUE TITLE
(Semi) automated black box testing

### ISSUE THREAD

**Issue by plamola** [magneticio/vamp#289]
**tldr;** *Create a test runner which can test the functionality of the complete Vamp stack, including different container drivers.*

To do proper regression / integration testing, we need to create a setup which enables us to test the components of Vamp. This feature proposal contains a number of steps to get a full set of tests up and running. Each step should be converted to a separate issue.

# 1. Create the basic test runner (#295)

The test runner should
- be pointed at a running vamp installation,
- ensure the test environment is clean,
- walk through test scenario's,
- confirm all the steps are performed,
- report the result of each test,
- clean up the environment after a run,
- report the aggregate state of all the test.

# 2. Create a test scenario for the docker tutorial on vamp.io (#335)

The tutorial on our website should always work. If it doesn't, either the tutorial needs to be changed, or the code. Create a test scenario for the test runner, which does all the necessary checks.
  
# 3. Create a 'test environment in a box'

This can be either a docker image, containing all 4 Vamp modules or an ansible script to provision a (set of) machine(s).

# 4. Automate test execution

Have Travis CI provision a test environment on a cloud platform and perform the tests automatically.

# 5. Expand the set of test scenarios

Test more scenarios, add additional checks for component states, support all container drivers etc.
This will be a work in progress.

**Comment by dragoslav** [comment]
This is also covered by https://github.com/magneticio/vamp-runner
